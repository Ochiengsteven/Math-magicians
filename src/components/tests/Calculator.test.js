import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
import Calculator from '../calculator';

describe('Calculator component', () => {
  it('should match the snapshot', () => {
    const { asFragment } = render(<Calculator />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should display the result of arithmetic operations', () => {
    const { getByText, queryAllByText } = render(<Calculator />);

    // Simulate button clicks to perform calculations
    userEvent.click(getByText('1'));
    userEvent.click(getByText('+'));
    userEvent.click(getByText('2'));
    userEvent.click(getByText('='));

    // Check if the result is displayed correctly
    const resultElement = queryAllByText('3', { selector: '.operation' });
    expect(resultElement).toHaveLength(1); // Ensure there's only one element with the text "3"
  });

  it('should clear the screen after AC button is clicked', () => {
    const { getByText, queryByText } = render(<Calculator />);

    // Simulate button clicks to enter a number and then clear
    userEvent.click(getByText('1'));
    userEvent.click(getByText('AC'));

    // Check if the screen container is cleared
    const screenContainer = queryByText('.operation');
    expect(screenContainer).toBeNull();
  });

  it('should match the snapshot after interactions', () => {
    const { asFragment, getByText } = render(<Calculator />);

    // Simulate button clicks to perform interactions
    userEvent.click(getByText('1'));
    userEvent.click(getByText('+'));
    userEvent.click(getByText('2'));
    userEvent.click(getByText('='));
    userEvent.click(getByText('AC'));

    // Assert that the snapshot matches the updated component state
    expect(asFragment()).toMatchSnapshot();
  });
});
