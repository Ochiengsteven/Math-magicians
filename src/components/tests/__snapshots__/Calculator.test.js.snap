// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator component should match the snapshot 1`] = `
<DocumentFragment>
  <div
    class="calculator-container"
  >
    <h1>
      Let's do some math!
    </h1>
    <div
      class="calculator"
    >
      <div
        class="output-screen"
      >
        <div
          class="operation"
        />
      </div>
      <div
        class="button-grid"
      >
        <button
          class="calc-button  "
          type="button"
        >
          AC
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          +/-
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          %
        </button>
        <button
          class="calc-button  operation-button"
          type="button"
        >
          /
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          7
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          8
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          9
        </button>
        <button
          class="calc-button  operation-button"
          type="button"
        >
          *
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          4
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          5
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          6
        </button>
        <button
          class="calc-button  operation-button"
          type="button"
        >
          -
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          1
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          2
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          3
        </button>
        <button
          class="calc-button  operation-button"
          type="button"
        >
          +
        </button>
        <button
          class="calc-button double-width "
          type="button"
        >
          0
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          .
        </button>
        <button
          class="calc-button  operation-button"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Calculator component should match the snapshot after interactions 1`] = `
<DocumentFragment>
  <div
    class="calculator-container"
  >
    <h1>
      Let's do some math!
    </h1>
    <div
      class="calculator"
    >
      <div
        class="output-screen"
      >
        <div
          class="operation"
        />
      </div>
      <div
        class="button-grid"
      >
        <button
          class="calc-button  "
          type="button"
        >
          AC
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          +/-
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          %
        </button>
        <button
          class="calc-button  operation-button"
          type="button"
        >
          /
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          7
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          8
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          9
        </button>
        <button
          class="calc-button  operation-button"
          type="button"
        >
          *
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          4
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          5
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          6
        </button>
        <button
          class="calc-button  operation-button"
          type="button"
        >
          -
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          1
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          2
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          3
        </button>
        <button
          class="calc-button  operation-button"
          type="button"
        >
          +
        </button>
        <button
          class="calc-button double-width "
          type="button"
        >
          0
        </button>
        <button
          class="calc-button  "
          type="button"
        >
          .
        </button>
        <button
          class="calc-button  operation-button"
          type="button"
        >
          =
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
